src/main/java/com/deloitte/elrr/aggregator/rules/ProcessAssigned.java

```diff
@@ -0,0 +1,287 @@
+import com.deloitte.elrr.entity.Credential;
+import com.deloitte.elrr.entity.Goal;
+import com.deloitte.elrr.entity.LearningResource;
+import com.deloitte.elrr.entity.Person;
+import com.deloitte.elrr.entity.types.GoalType;
+import com.deloitte.elrr.exception.RuntimeServiceException;
+import com.deloitte.elrr.jpa.svc.GoalSvc;
+import com.yetanalytics.xapi.model.AbstractActor;
+import com.yetanalytics.xapi.model.Activity;
+import com.yetanalytics.xapi.model.Context;
+import com.yetanalytics.xapi.model.Statement;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class ProcessAssigned implements Rule {
+
+    @Autowired
+    private ProcessPerson processPerson;
+
+    @Autowired
+    private LangMapUtil langMapUtil;
+
+    @Autowired
+    private GoalSvc goalService;
+
+    @Autowired
+    private LearningResourceUtil learningResourceUtil;
+
+    @Autowired
+    private ProcessCredential processCredential;
+
+    @Autowired
+    private ProcessCompetency processCompetency;
+
+    @Autowired
+    private ExtensionsUtil extensionsUtil;
+
+    private static final String GOAL_MESSAGE = "Goal";
+
+    /**
+     * @param statement
+     * @return boolean
+     */
+    public boolean fireRule(Statement statement) {
+
+        // If not an activity
+        if (!(statement.getObject() instanceof Activity)) {
+            return false;
+        }
+
+        // Is Verb Id = assigned
+        return (statement.getVerb().getId().toString().equalsIgnoreCase(
+                VerbIdConstants.ASSIGNED_VERB_ID.toString()));
+    }
+
+    /**
+     * @param person
+     * @param statement
+     * @return person
+     * @throws AggregatorException
+     * @throws URISyntaxException
+     */
+    public Person processRule(Person person, Statement statement)
+            throws AggregatorException, ClassCastException,
+            NullPointerException, RuntimeServiceException, URISyntaxException {
+
+        log.info("Process assigned.");
+
+        // Get start date
+        // Convert from ZonedDateTime to LocalDateTime
+        LocalDateTime startDate = statement.getTimestamp().toLocalDateTime();
+
+        // Get Activity
+        Activity activity = (Activity) statement.getObject();
+
+        // Get assigned actor
+        AbstractActor assignedActor = (AbstractActor) extensionsUtil
+                .getExtensions(statement.getContext(), "Actor");
+
+        // Process assigned person
+        Person assignedPerson = processPerson.processAssignedPerson(
+                assignedActor);
+
+        // Process Goal
+        processGoal(statement.getContext(), activity, startDate,
+                assignedPerson);
+
+        return person;
+
+    }
+
+    /**
+     * @param context
+     * @param activity
+     * @param startDate
+     * @param assignedPerson
+     * @return goal
+     * @throws AggregatorException
+     * @throws URISyntaxException
+     */
+    @Transactional
+    @SuppressWarnings("checkstyle:linelength")
```

Comment:
what is this needed for? If it's those long extensions check after the correction i made above.